File: api/internal/model/photo.go
package model

import (
	"time"
)

type Photo struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	Title        string    `gorm:"size:200;not null" json:"title"`
	Description  string    `gorm:"type:text" json:"description"`
	ImageURL     string    `gorm:"size:500;not null" json:"imageUrl"`
	ThumbnailURL string    `gorm:"size:500" json:"thumbnailUrl"`
	Category     string    `gorm:"size:50" json:"category"`
	Location     string    `gorm:"size:200" json:"location"`
	IsFeatured   bool      `gorm:"default:false" json:"isFeatured"`
	DisplayOrder int       `gorm:"default:0;index" json:"displayOrder"`
	Status       string    `gorm:"size:20;default:'draft';index" json:"status"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

type CreatePhotoRequest struct {
	Title        string `json:"title" binding:"required,min=1,max=200"`
	Description  string `json:"description"`
	ImageURL     string `json:"imageUrl" binding:"required"`
	ThumbnailURL string `json:"thumbnailUrl"`
	Category     string `json:"category"`
	Location     string `json:"location"`
	IsFeatured   bool   `json:"isFeatured"`
}

type UpdatePhotoRequest struct {
	Title        *string `json:"title"`
	Description  *string `json:"description"`
	ImageURL     *string `json:"imageUrl"`
	ThumbnailURL *string `json:"thumbnailUrl"`
	Category     *string `json:"category"`
	Location     *string `json:"location"`
	IsFeatured   *bool   `json:"isFeatured"`
	Status       *string `json:"status"`
}

File: web/lib/api/photo.ts
import { request } from "./base";

/**
 * Photo 相关 API
 */

export interface Photo {
  id: number;
  title: string;
  description: string;
  imageUrl: string;
  thumbnailUrl: string;
  category: string;
  location: string;
  isFeatured: boolean;
  displayOrder: number;
  status: string;
  createdAt: string;
  updatedAt: string;
}

interface ListPhotosResponse {
  data: Photo[];
  total: number;
}

interface CreatePhotoRequest {
  title: string;
  description?: string;
  imageUrl: string;
  thumbnailUrl?: string;
  category?: string;
  location?: string;
  isFeatured?: boolean;
}

interface UpdatePhotoRequest {
  title?: string;
  description?: string;
  category?: string;
  location?: string;
  isFeatured?: boolean;
  status?: string;
}

interface ReorderRequest {
  orders: Array<{ id: number; displayOrder: number }>;
}

/**
 * 获取所有照片
 */
export async function listPhotos(): Promise<ListPhotosResponse> {
  return request<ListPhotosResponse>("/photos");
}

/**
 * 获取单个照片
 */
export async function getPhotoById(id: number): Promise<Photo> {
  return request<Photo>(`/photos/${id}`);
}

/**
 * 创建照片
 */
export async function createPhoto(photo: CreatePhotoRequest): Promise<Photo> {
  return request<Photo>("/photos", {
    method: "POST",
    body: JSON.stringify(photo),
  });
}

/**
 * 更新照片
 */
export async function updatePhoto(id: number, photo: UpdatePhotoRequest): Promise<Photo> {
  return request<Photo>(`/photos/${id}`, {
    method: "PUT",
    body: JSON.stringify(photo),
  });
}

/**
 * 删除照片
 */
export async function deletePhoto(id: number): Promise<{ message: string }> {
  return request<{ message: string }>(`/photos/${id}`, {
    method: "DELETE",
  });
}

/**
 * 批量更新排序
 */
export async function reorderPhotos(
  orders: Array<{ id: number; displayOrder: number }>
): Promise<{ message: string }> {
  return request<{ message: string }>("/photos/reorder", {
    method: "POST",
    body: JSON.stringify({ orders }),
  });
}
